Modules/Core/Common/test/itkImageTest.cxx

```diff
@@ -234,5 +233,16 @@ itkImageTest(int, char *[])
     return EXIT_FAILURE;
   }
 
+  // Test that a const pointer can be instantiated from a non-const pointer
+  Image::ConstPointer myconstptr = image;
+  if (!image->IsCongruentImageGeometry(myconstptr, 0.0, 0.0))
+  {
+    std::cerr << "Image compare to self fails IsCongruentImageGeometry" << std::endl;
+  }
+  if (!image->IsSameImageGeometryAs(myconstptr, 0.0, 0.0))
+  {
+    std::cerr << "Image compare to self fails IsSameImageGeometryAs" << std::endl;
+  }
+
   return (EXIT_SUCCESS);
```

Comment:
Technically, your test does not fail when IsCongruentImageGeometry returns false! It reaches `return (EXIT_SUCCESS)` anyway.  

Please 🙏 consider using GoogleTest for tests like this. With GTest, it would just be:

    EXPECT_TRUE(image->IsSameImageGeometryAs(myconstptr, 0.0, 0.0));

Reply:
I just added these methods for SimpleITK and took full advantage of things for testing:
https://github.com/SimpleITK/SimpleITK/pull/2097/files#diff-b59e3fb36168b499154d941782889660f04ab0c66ed5a31f4a994ca88764b17d

@hjmjohnson  I'llI see what I can do in the itkImageGTest.cxx file.